lib/huginn_acumen_order_agent/acumen_order_agent.rb

```diff
@@ -0,0 +1,152 @@
+        forward processing the next bundle.
+
+        For example, if this agent receives two products as input (`A` and `B`), and we fail to
+        load the Inv_Product record for product `A`, the agent would emit an error payload of:
+
+        ```
+        {
+          status: 500,
+          scope: 'Fetch Inv_Product Data',
+          message: 'Failed to lookup Inv_Product record for Product A',
+          data: { product_id: 123 },
+          trace: [ ... ]
+        }
+        ```
+
+        The goal of this approach is to ensure the agent outputs as much data as reasonably possible
+        with each execution. If there is an error in the Paperback version of a title, that shouldn't
+        prevent this agent from returning the Hardcover version.
+
+        MD
+
+        def default_options
+            {
+                'endpoint' => 'https://example.com',
+                'site_code' => '',
+                'password' => '',
+                'digital_formats' => [],
+                'attribute_to_property' => {},
+            }
+        end
+
+        def validate_options
+            unless options['endpoint'].present?
+                errors.add(:base, 'endpoint is a required field')
+            end
+
+            unless options['site_code'].present?
+                errors.add(:base, 'site_code is a required field')
+            end
+
+            unless options['password'].present?
+                errors.add(:base, 'password is a required field')
+            end
+
+
+        end
+
+        def working?
+            received_event_without_error?
+        end
+
+        def check
+            handle interpolated['payload'].presence || {}
+        end
+
+        def receive(incoming_events)
+            incoming_events.each do |event|
+                handle(event)
+            end
+       end
+
+        private
+
+        def handle(event)
+            # Process agent options
+            endpoint = interpolated['endpoint']
+            site_code = interpolated['site_code']
+            password = interpolated['password']
+
+            # Configure the Acumen Client
+            auth = {
+                'site_code' => site_code,
+                'password' => password,
+                'endpoint' => endpoint,
+            }
+            client = AcumenOrderClient.new(faraday, auth)
+
+            order_codes = event.payload['order_codes']
+            log('------------------------------------------------------------')
+                log(":: order codes #{order_codes}")
+            log('------------------------------------------------------------')
+
+            begin
+              invoices = fetch_invoice_data(client, order_codes)
+
+              unless invoices.blank?
+                log('------------------------------------------------------------')
+                    log(":: invoices #{invoices}")
+                log('------------------------------------------------------------')
```

Comment:
Do we want to keep these logs? Same with logs on 118, 119, and 120?

Reply:
No we do not. Fixed